### ISSUE TITLE
NativeFramePresenter does not set the parent DO on pages

### ISSUE THREAD

**Issue by dr1rrb** [unoplatform/uno.toolkit.ui#449]
## Current behavior
On iOS the `NativeFramePresenter` does not set parent `DependencyObject` of the `Page` that it inject in the visual tree. 

This breaks visual tree navigation using the `ViusualTreeHelper`.

## Expected behavior
`NativeFramePresenter` should set itself as parent `DependencyObject` of the `Page`, ignoring the native only element.

## How to reproduce it (as minimally and precisely as possible)
Use `Frame` whith **material** theme and browse the tree using the `VisualTreeHelper`

## Environment
Nuget Package: Uno.Toolkit.UI
Package Version(s): __latest__
Affected platform(s):
- [x] iOS
- [?? most probably??] Android
- [ ] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks

**Comment by kazo0** [comment]
@dr1rrb and this is working with the NativeFramePresenter coming from Uno instead?

**Comment by dr1rrb** [comment]
It's working when using the default style of the `Frame` from Uno. Not sure it's using the the `NativeFramePresenter`, but if so, yes ^^

**Comment by jeromelaban** [comment]
The default `Frame` is using all managed controls for displaying content, that does explain the difference.

**Comment by kazo0** [comment]
Yeah I believe setting `FeatureConfiguration.Style.ConfigureNativeFrameNavigation()` on a plain old vanilla Uno application would probably result in the same issue
